frameworks/Elixir/phoenix/lib/hello_web/controllers/page_controller.ex

```diff
@@ -48,54 +44,51 @@ defmodule HelloWeb.PageController do
 
     fortunes =
       [additional_fortune | Repo.all(Fortune)]
-      |> Enum.sort_by(& &1.message)
+      |> Enum.sort(fn a, b -> a.message < b.message end)
 
     render(conn, :fortunes, fortunes: fortunes)
   end
 
   def updates(conn, params) do
-    {:ok, worlds} =
-      Repo.transaction(fn ->
-        :rand.seed(:exsp)
-
-        worlds =
-          Stream.repeatedly(&random_id/0)
-          |> Stream.uniq()
-          |> Stream.map(&Repo.get(World, &1))
-          |> Stream.map(fn world -> %{id: world.id, randomnumber: :rand.uniform(@random_max)} end)
-          |> Enum.take(size(params["queries"]))
-          # If this is not sorted it sometimes generates
-          #  FAIL for http://tfb-server:8080/updates/20
-          #  Only 20470 executed queries in the database out of roughly 20480 expected.
-          |> Enum.sort_by(& &1.id)
-
-        Repo.insert_all(
-          World,
-          worlds,
-          on_conflict: {:replace_all_except, [:id]},
-          conflict_target: [:id],
-          returning: false
-        )
-
-        worlds
+    world_updates =
+      Repo.checkout(fn ->
+        params["queries"]
+        |> random_ids_sample()
+        |> Enum.sort()
+        #
+        # If this is not sorted it will intermittently generate:
+        #
+        #   FAIL for http://tfb-server:8080/updates/20
+        #   Only 20470 executed queries in the database out of roughly 20480 expected.
+        #
+        |> Enum.map(fn id ->
+          world = Repo.get(World, id)
+          %{id: world.id, randomnumber: :rand.uniform(@random_max)}
+        end)
       end)
 
-    json(conn, worlds)
+    Repo.insert_all(
+      World,
+      world_updates,
+      on_conflict: {:replace_all_except, [:id]},
+      conflict_target: [:id],
+      returning: false
+    )
+
+    json(conn, world_updates)
   end
 
   def plaintext(conn, _params) do
     text(conn, "Hello, World!")
```

Comment:
`text/2` includes "charset=utf-8" in the content type and the other benchmarks seem to not do it (and this will matter on the plain text one numbers). So it probably makes sense for us to set the content-type explicitly instead:

```suggestion
    conn
    |> put_resp_header("content-type", "text/plain")
    |> send_resp(200, "Hello, World!")
```

WDYT?

Reply:
Makes sense to me.  Adding that now.